vavr/src/main/java/io/vavr/Value.java

```diff
@@ -657,27 +699,76 @@ default CharSeq toCharSeq() {
     }
 
     /**
-     * Converts this to an {@link java.util.List}.
+     * Converts this to a mutable {@link java.util.List}.
+     * Elements are added by calling {@link java.util.List#add(Object)}.
+     *
+     * <pre>{@code
+     * // = []
+     * Future.<String> of(() -> { throw new Error(); })
+     *       .toJavaList()
+     * 
+     * // = [ok]
+     * Try.of(() -> "ok")
+     *    .toJavaList()
+     *
+     * // = [1, 2, 3]
+     * List.of(1, 2, 3)
+     *     .toJavaList()
+     * }</pre>
      *
      * @return A new {@link java.util.ArrayList}.
      */
     default java.util.List<T> toJavaList() {
-        return ValueModule.toJavaCollection(this, ArrayList::new);
+        return ValueModule.toJavaCollection(this, ArrayList::new, 10);
```

Comment:
Why?

Reply:
Before we used a default initial size of 16. But Java's ArrayList has 10 (hidden in a private field). I think we should use the default as fallback in order to be 'as good as Java'.

<img width="520" alt="screen shot 2017-09-12 at 19 39 30" src="https://user-images.githubusercontent.com/743833/30340291-54d1d3f6-97f2-11e7-8a69-63132f0522f2.png">

(_Note: I used IntelliJ IDEA to look into java.util.ArrayList of JDK 1.8.0_144_)

That the fallback value of 10 will only be taken in the following cases:

* We have a single-valued `Value`, like `Option` or `Try`. Please note that we do not use an initial size of 1 for single-valued type. This is because we return a mutable collection that might be modified by the caller (this is absolutely legitimate!). If the ArrayList has to be resized, it would be counter-productive/too costly to have a size of 1. Therefore we take the default size of 10.
* Another case, where 10 is taken as initial capacity, is for collections that are traversable once (like Iterator) or when they are lazy (like Stream). The latter case fixes the original issue #2020.